ql/src/java/org/apache/hadoop/hive/ql/parse/EximUtil.java

```diff
@@ -300,10 +300,10 @@ public static void createExportDump(FileSystem fs, Path metadataPath, Table tabl
     }
 
     try (JsonWriter writer = new JsonWriter(fs, metadataPath)) {
+      new TableSerializer(tableHandle, partitions, hiveConf).writeTo(writer, replicationSpec);
```

Comment:
Why is it moved up?

Reply:
replication spec is modified in the serializer to avoid iterating over the partitions again. So replication spec dump is moved down.